app/src/lib/stores/app-store.ts

```diff
@@ -3347,44 +3348,102 @@ export class AppStore extends TypedBaseStore<IAppState> {
   public async _deleteBranch(
     repository: Repository,
     branch: Branch,
-    includeRemote: boolean
+    includeRemote?: boolean
   ): Promise<void> {
     return this.withAuthenticatingUser(repository, async (r, account) => {
-      const { branchesState } = this.repositoryStateCache.get(r)
-      let branchToCheckout = branchesState.defaultBranch
-
-      // If the default branch is null, use the most recent branch excluding the branch
-      // the branch to delete as the branch to checkout.
-      if (branchToCheckout === null) {
-        let i = 0
-
-        while (i < branchesState.recentBranches.length) {
-          if (branchesState.recentBranches[i].name !== branch.name) {
-            branchToCheckout = branchesState.recentBranches[i]
-            break
-          }
-          i++
-        }
+      const gitStore = this.gitStoreCache.get(r)
+
+      // If soley a remote branch, there is no need to checkout a branch.
```

Comment:
Nit: typo? (you can fix it in the other PR if nothing else comes up in this one, don't worry 😆 )
```suggestion
      // If solely a remote branch, there is no need to checkout a branch.
```